pymc/distributions/multivariate.py

```diff
@@ -2192,33 +2192,24 @@ def make_node(self, rng, size, dtype, alpha, K):
         alpha = at.as_tensor_variable(alpha)
         K = at.as_tensor_variable(intX(K))
 
-        if alpha.ndim > 0:
-            raise ValueError("The concentration parameter needs to be a scalar.")
-
         if K.ndim > 0:
             raise ValueError("K must be a scalar.")
 
         return super().make_node(rng, size, dtype, alpha, K)
 
-    def _infer_shape(self, size, dist_params, param_shapes=None):
-        alpha, K = dist_params
-
-        size = tuple(size)
-
-        return size + (K + 1,)
+    def _supp_shape_from_params(self, dist_params, **kwargs):
+        K = dist_params[1]
+        return (K + 1,)
 
     @classmethod
     def rng_fn(cls, rng, alpha, K, size):
         if K < 0:
             raise ValueError("K needs to be positive.")
 
-        if size is None:
-            size = (K,)
-        elif isinstance(size, int):
-            size = (size,) + (K,)
-        else:
-            size = tuple(size) + (K,)
+        size = to_tuple(size)
+        size = np.broadcast_shapes(alpha.shape, size) + (K,)
```

Comment:
This is still a bit incorrect, alpha is not allowed to broadcast size. Simply put `size = size if size is not None else alpha.shape`

Reply:
Ok, Thank you

Reply:
> alpha is not allowed to broadcast size

Hi @ricardoV94 

Considering the below snippet and assuming MvNormal works with batched covariance matrix, what is shape of resulting MvNormal:

```python
mu = np.array([1, 2, 3])
cov = np.eye(3)
batched_cov = np.full((10, 3, 3), cov)
pm.MvNormal.dist(mu=mu, cov=cov, size=(5, 1)).eval().shape
```

Will it be (5, 1, 10, 3) or (5, 10, 3)?

I think the shape should be (5, 10, 3) as batches from alpha can broadcast with size.

Reply:
That's invalid. Size determines the number of samples. It can either be implied by the batches dimensions of the parameters, match them or go beyond.

It should behave like `np.full(value, shape)` (I don't remember the order of the parameters). You can't ask for `np.full(np.zeros((10, 3, 3), size=(5, 1))`

Reply:
Deleted by mistake. The example is invalid, it should fail. Size can be either `None`, or some tuple that ends in `(10, 3)`. It can't contradict the shape already implied by the covariance. Similar to how `np.full` where the shape can't broadcast to match the value, only the other way around.

Reply:
Got it. Thanks Ricardo. 

I think this behaviour (that size can't contradict the batched shape already implied) is a lot easier to work with, compared to PyMC v3.